src/utils/mdlUpgrade.js

```diff
@@ -1,19 +1,9 @@
 import React from 'react';
+import MDLComponent from './MDLComponent';
 
-export default (Component) => {
-    class MDLUpgradedComponent extends React.Component {
-        componentDidMount() {
-            componentHandler.upgradeElement(React.findDOMNode(this));
-        }
-
-        componentWillUnmount() {
-            componentHandler.downgradeElements(React.findDOMNode(this));
-        }
-
-        render() {
-            return <Component {...this.props} />;
-        }
+export default Component =>
+class MDLUpgradedComponent extends React.Component {
+    render() {
+        return <MDLComponent><Component {...this.props} /></MDLComponent>;
```

Comment:
So for components using this helper, there are now 2 indirections:
```
MDLUpgradedComponent
    MDLComponent
        Component
```
While if we reverse the helper function and component, we'll have only 1 indirections in most cases.